grpc-native/src/main/java/org/ballerinalang/net/grpc/builder/syntaxtree/SyntaxTreeGen.java

```diff
@@ -0,0 +1,302 @@
+        TextDocument textDocument = TextDocuments.from("");
+        SyntaxTree syntaxTree = SyntaxTree.from(textDocument);
+        return syntaxTree.modifyWith(modulePartNode);
+    }
+
+    public static SyntaxTree generateSyntaxTree(ServiceStub serviceStub, String mode) {
+        NodeList<ModuleMemberDeclarationNode> moduleMembers = AbstractNodeFactory.createEmptyNodeList();
+        NodeList<ImportDeclarationNode> imports = AbstractNodeFactory.createEmptyNodeList();
+
+        if (GRPC_SERVICE.equals(mode)) {
+            ImportDeclarationNode importForGrpc = Imports.getImportDeclarationNode("ballerina", "grpc");
+            imports = AbstractNodeFactory.createNodeList(importForGrpc);
+
+            Listener listener = new Listener(
+                    "ep",
+                    getQualifiedNameReferenceNode("grpc", "Listener"),
+                    getImplicitNewExpressionNode(new String[]{"9090"}),
+                    false
+            );
+            moduleMembers = moduleMembers.add(listener.getListenerDeclarationNode());
+
+            Service service = new Service(
+                    new String[]{"\"" + serviceStub.getServiceName() + "\""},
+                    new String[]{"ep"}
+            );
+            Annotation grpcServiceDescriptor = new Annotation("grpc", "ServiceDescriptor");
+            grpcServiceDescriptor.addField("descriptor", "ROOT_DESCRIPTOR");
+            grpcServiceDescriptor.addField(
+                    "descMap",
+                    getFunctionCallExpressionNode("getDescriptorMap", new String[]{})
+            );
+            service.addAnnotation(grpcServiceDescriptor.getAnnotationNode());
+
+            List<Method> methodList = new ArrayList<>();
+            methodList.addAll(serviceStub.getUnaryFunctions());
+            methodList.addAll(serviceStub.getClientStreamingFunctions());
+            methodList.addAll(serviceStub.getServerStreamingFunctions());
+            methodList.addAll(serviceStub.getBidiStreamingFunctions());
+
+            for (Method method : methodList) {
+                Function function = new Function(method.getMethodName());
+                String input = method.getInputType();
+                String output = method.getOutputType();
+                function.addRequiredParameter(getSimpleNameReferenceNode(input), "value");
+                function.addReturns(
+                        getUnionTypeDescriptorNode(
+                                getStreamTypeDescriptorNode(
+                                        getSimpleNameReferenceNode(output),
+                                        // Todo: parametarize identirifer (optionalTypeDescriptor)
+                                        getOptionalTypeDescriptorNode("", "error")
+                                ),
+                                getErrorTypeDescriptorNode()
+                        )
+                );
```

Comment:
input and output types are deferred by the message pattern. e.g: in client streaming, input param is a stream type, in server streaming output type is a stream type

Reply:
Fixed in 9d8e1df